packages/logs/src/domain/logger.ts

```diff
@@ -64,22 +60,42 @@ export class Logger {
   }
 
   setContext(context: object) {
-    this.sender.getContextManager().set(context)
+    this.options.contextManager.set(context)
   }
 
   addContext(key: string, value: any) {
-    this.sender.getContextManager().add(key, value)
+    this.options.contextManager.add(key, value)
   }
 
   removeContext(key: string) {
-    this.sender.getContextManager().remove(key)
+    this.options.contextManager.remove(key)
   }
 
   setHandler(handler: HandlerType | HandlerType[]) {
-    this.sender.setHandler(handler)
+    this.options.handlerType = handler
   }
 
   setLevel(level: StatusType) {
-    this.sender.setLevel(level)
+    this.options.level = level
+  }
+}
+
+export type LoggerOptions = ReturnType<typeof newLoggerOptions>
+
+export function newLoggerOptions(
+  name?: string,
+  handlerType: HandlerType | HandlerType[] = HandlerType.http,
+  level: StatusType = StatusType.debug,
+  context: object = {}
+) {
+  const contextManager = createContextManager()
+  // Todo: merge the logger name into the context in logger collection in next major version
+  contextManager.set(assign({}, context, { logger: { name } }))
+
+  return {
+    handlerType,
+    level,
+    name,
+    contextManager,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I think we should handle separately options used to build the current message (name, context) from the one used to "handle" this message (handler, level).

We could partially build the message inside the logger like we did before:
```ts
combine({ message, status }, this.contextManager.get(), messageContext)
```

and let that only override more general contexts (global, rum, service/env/version) during assembly.

wdyt?